recipes/mosquitto/1.x/conanfile.py

```diff
@@ -0,0 +1,132 @@
+        extracted_dir = self.name.replace("-", ".") + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["WITH_PIC"] = self.options.get_safe("fPIC", True)
+            self._cmake.definitions["WITH_STATIC_LIBRARIES"] = not self.options.shared
+            self._cmake.definitions["WITH_TLS"] = self.options.with_tls
+            self._cmake.definitions["WITH_TLS_PSK"] = self.options.with_tls
+            self._cmake.definitions["WITH_EC"] = self.options.with_tls
+            self._cmake.definitions["DOCUMENTATION"] = False
+            self._cmake.definitions["WITH_THREADING"] = self.settings.compiler != "Visual Studio"
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        if self.settings.os == "Windows":
+            if self.options.with_tls:
+                tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                    "set (LIBRARIES ${LIBRARIES} ws2_32)",
+                                    "set (LIBRARIES ${LIBRARIES} ws2_32 crypt32)")
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                                    "set (MOSQ_LIBS ${MOSQ_LIBS} ws2_32)",
+                                    "set (MOSQ_LIBS ${MOSQ_LIBS} ws2_32 crypt32)")
+                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "CMakeLists.txt"),
+                                    "target_link_libraries(mosquitto_passwd ${OPENSSL_LIBRARIES})",
+                                    "target_link_libraries(mosquitto_passwd ${OPENSSL_LIBRARIES} ws2_32 crypt32)")
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "CMakeLists.txt"),
+                                "install(TARGETS libmosquitto RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")",
+                                "install(TARGETS libmosquitto RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\" ARCHIVE DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")")
+            tools.replace_in_file(os.path.join(self._source_subfolder, "lib", "cpp", "CMakeLists.txt"),
+                                "install(TARGETS mosquittopp RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")",
+                                "install(TARGETS mosquittopp RUNTIME DESTINATION \"${CMAKE_INSTALL_BINDIR}\" LIBRARY DESTINATION \"${CMAKE_INSTALL_LIBDIR}\" ARCHIVE DESTINATION \"${CMAKE_INSTALL_LIBDIR}\")")
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
```

Comment:
Is possible, it will be better for future maintenance to have alike 1.x and 2.x methods (package, package_id).

Reply:
No because some of the app and the separation of the c and c++ apis are not possible... the CMake options upstream changed significantly.